content/docs/04-ai-sdk-ui/03-chatbot-message-persistence.mdx

```diff
@@ -420,9 +423,46 @@ export async function GET(request: Request) {
     execute: () => {},
   });
 
-  return new Response(
-    await streamContext.resumableStream(recentStreamId, () => emptyDataStream),
+  const stream = await streamContext.resumableStream(
+    recentStreamId,
+    () => emptyDataStream,
   );
+
+  /*
+   * For when the generation is "active" during SSR but the
+   * resumable stream has concluded after reaching this point.
+   */
+  if (!stream) {
+    const messages = await getMessagesByChatId({ id: chatId });
+    const mostRecentMessage = messages.at(-1);
+
+    if (!mostRecentMessage) {
+      return new Response(emptyDataStream, { status: 200 });
+    }
+
+    if (mostRecentMessage.role !== 'assistant') {
+      return new Response(emptyDataStream, { status: 200 });
+    }
+
+    const messageCreatedAt = new Date(mostRecentMessage.createdAt);
+
+    if (differenceInSeconds(resumeRequestedAt, messageCreatedAt) > 15) {
```

Comment:
what is the goal here, is it a heuristic to determine if the proverbial ship has sailed? feels worth commenting on goal and what 15 represents

Reply:
If the stream has ended, we send the assistant message generated as a SSE to the client to resolve. The goal here is to make that event only available for 15 seconds instead of it being persistent forever. It is worth noting that there is no drawback if we remove this check – do you think we should keep it or not?

I can comment the reasoning if we do decided to keep it.

Reply:
I gave it some thought and figured it'd be best to remove it since it introduces more complexity to the example snippet. Will revisit once we receive feedback about this.

Reply:
Sounds good, I'd say take it out too for now.